pinot-controller/src/main/java/org/apache/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -729,6 +729,24 @@ public List<String> getAllOfflineTables() {
     return offlineTableNames;
   }
 
+  /**
+   * Get all dimension table names.
+   *
+   * @return List of dimension table names
+   */
+  public List<String> getAllDimensionTables() {
+    List<String> dimensionTableNames = new ArrayList<>();
+    for (String resourceName : getAllResources()) {
+      if (TableNameBuilder.isOfflineTableResource(resourceName)) {
+        final TableConfig tableConfig = getTableConfig(resourceName);
```

Comment:
This will read table config for all tables (potentially thousands) which can be very expensive. Are we sure we want to support this? A cheaper way would be to read it from `TableCache`

Reply:
This will provide users easy way to get the list of dimension tables, instead of getting all the configs and filtering them. Changed the code to use TableCache.